### ISSUE TITLE
Helper Function to determine resolution of coordinates

### ISSUE THREAD

**Issue by mpu-creare** [creare-com/podpac#502]
**Description**
When working with lat/lon or other projected coordinates, it's hard to know the real nominal resolution of your coordinate system.

**Describe the solution you'd like**
I want a helper function on coordinates that give me the resolution. 

**Sample Code**
``` python
>>> coords = podpac.Coordinates([...], dims=['lat', 'lon']
>>> coords.horizontal_resolution(units='m', type='nominal')
{"lat": 10,
"lon": 20, 
"units": "m"
}
```

**Comment by mpu-creare** [comment]
This code might be useful: https://github.com/creare-com/pydem/blob/master/pydem/utils.py#L145

**Comment by mpu-creare** [comment]
This might also be useful: https://pyproj4.github.io/pyproj/stable/api/crs/crs.html

**Comment by CFoye-Creare** [comment]
Left to-do
- [ ] Finish Unit Test
- [ ] Write code to convert degrees to metres
- [ ] Stacked Conversion
- [ ] Unstacked Conversion
- [ ] Use podpac.units implementation

**Comment by CFoye-Creare** [comment]
- [ ] cKDTree with boxsize wraparound implementation for stacked NearestNeighbor

**Comment by mpu-creare** [comment]
For `coords.horizontal_resolution(units='m', type='nominal')`, let's talk about the different types

types = 
* "nominal" <-- this is wrong but cheap to calculate. Give a 'nominal' resolution over the entire domain
* "summary" <-- This is still approximate, gives the mean and standard deviation to each point
* "full" <-- min distance between every point (dx, dy, basically)
* "full_exact" <-- uses brute-force for the stacked coordinates

This is a proposal, feel free to propose something better.

**Comment by CFoye-Creare** [comment]
Implemented 3 of the 4 types:

- "nominal" <-- this is wrong but cheap to calculate. Give a 'nominal' resolution over the entire domain
- "summary" <--- Returns average and std.dev for each point.
- "full" <--- returns the exact distance between every point, or every dx,dy if coordinates are unstacked

**Comment by CFoye-Creare** [comment]
9851b1cb5ce4b32bc00e96e500ffc8c78efd7376

**Comment by CFoye-Creare** [comment]
e05a662688b5d600aeb29784d43b5acb8c466cbe

**Comment by CFoye-Creare** [comment]
Wrote some code to address the problem of lat/lon evaluating to different distances based upon the other dimension's value. However, I realized once passing in non-square unstacked coordinates that I failed to account for the difference resolution changing for each differing value of the other dimension.

Let our unstacked coordinate system be a $M$ by $N$ matrix, where $M$ is the number of rows (latitude values), and $N$ is the number of columns (longitude values). See Blender visualization:

![image](https://user-images.githubusercontent.com/122571842/218126287-b6833a29-c054-4955-b339-c25a5d1cfe51.png)

Previously, we were returning two arrays of size $M$ and size $N$. This meant we were returning either a single row/column or a selection of each column. For example:
![image](https://user-images.githubusercontent.com/122571842/218126693-232f4944-1b34-40c7-af49-4f32f4900cbc.png)
Another example:
![image](https://user-images.githubusercontent.com/122571842/218126767-8b842bca-98d8-4e5e-a986-5f06c082f45c.png)

Instead, we want to return this for latitude:
![image](https://user-images.githubusercontent.com/122571842/218127220-46f1e717-74ef-431c-bad5-b374072b6735.png)

So, for latitude we want to return a $M+1$ by $N$ matrix.

For longitude:
![image](https://user-images.githubusercontent.com/122571842/218127535-f952f84e-5366-4c4b-9144-6a833dd99b06.png)
So, for longitude we want to return a $M$ by $N+1$ matrix.

For the "summary" type resolution, we can simply average/compute std.dev of all of these values.

**Comment by CFoye-Creare** [comment]
The $M+1$ or $N+1$ implementation is not possible due to the fact we don't store the values in the +1 dimension in our coords.

**Comment by mpu-creare** [comment]
closed by #505
